tests/protocol/mock_wire_protocol.py

```diff
@@ -28,48 +28,56 @@ def create(mock_wire_data_file):
     Creates a mock WireProtocol object that will return the data specified by 'mock_wire_data_file' (which must
     follow the structure of the data files defined in tests/protocol/mockwiredata.py).
 
-    NOTE: This function creates mocks for azurelinuxagent.common.utils.restutil.http_get and
+    NOTE: This function creates mocks for azurelinuxagent.common.utils.restutil.http_request and
```

Comment:
I had to mock 1 level deeper (http_request instead of http_get) to address all the scenarios in the tests,